test/e2e/common.go

```diff
@@ -629,8 +629,12 @@ func MachineToIPAddress(ctx context.Context, cli client.Client, m *clusterv1.Mac
 }
 
 func runCommand(logFolder, filename, machineIP, user, command string) error {
-	home := os.Getenv("HOME")
-	privkey, err := os.ReadFile(path.Join(home, "/.ssh/id_rsa"))
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("could not get home directory: %w", err)
+	}
+	key_path := path.Join(filepath.Clean(home), ".ssh", "id_rsa")
+	privkey, err := os.ReadFile(key_path) //#nosec
```

Comment:
gosec is still giving G304 here. Variable no longer is tainted via `$HOME` and is cleaned to be safe. Any ideas besides `//#nosec`? @lentzi90

Reply:
No idea. Is there even a situation where it would not complain unless a constant is used? :shrug:

Reply:
Apparently it wants constant, and no matter how much it is validated it is not going to be happy. Ignored it.